### PR TITLE
Update UMF and add fetching CUDA

### PR DESCRIPTION
**Pull request by ldorau**
It contains:
"Fix fetching L0 and CUDA headers and add UMF_CUDA_INCLUDE_DIR"
https://github.com/oneapi-src/unified-memory-framework/pull/832

### CODE REVIEW

**Comment by ldorau** [comment]
@pbalcer @bratpiorka please review

File: source/common/CMakeLists.txt [hunk 1]
```diff
@@ -9,6 +9,12 @@ if (UR_BUILD_ADAPTER_L0 OR UR_BUILD_ADAPTER_L0_V2)
     set(UMF_LEVEL_ZERO_INCLUDE_DIR "${LEVEL_ZERO_INCLUDE_DIR}" CACHE INTERNAL "Level Zero headers")
 endif()
 
+if(UR_BUILD_ADAPTER_CUDA)
+    include(FetchCUDA)
+    set(UMF_BUILD_CUDA_PROVIDER ON CACHE INTERNAL "Build UMF CUDA provider")
+    set(UMF_CUDA_INCLUDE_DIR "${CUDA_INCLUDE_DIR}" CACHE INTERNAL "CUDA headers")
+endif()
```
Comment (PatKamin) [inline source/common/CMakeLists.txt:16]: `else UMF_BUILD_CUDA_PROVIDER OFF` as it is `ON` by default in UMF.

File: cmake/FetchCUDA.cmake [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+    endif()
+
+    message(STATUS "CUDA Adapter: Will fetch CUDA ${UR_CUDA_TAG} from ${UR_CUDA_REPO}")
+    include(FetchContent)
+    FetchContent_Declare(cuda-content
+        GIT_REPOSITORY    ${UR_CUDA_REPO}
+        GIT_TAG           ${UR_CUDA_TAG}
+    )
+    FetchContent_MakeAvailable(cuda-content)
+    FetchContent_GetProperties(cuda-content)
```
Comment (PatKamin) [inline cmake/FetchCUDA.cmake:42]: Redundant. `MakeAvailable` handles the same variables.

File: source/common/CMakeLists.txt (hunk 1 above)
Comment (ldorau) [reply source/common/CMakeLists.txt:16]: Done

File: cmake/FetchCUDA.cmake (hunk 2 above)
Comment (ldorau) [reply cmake/FetchCUDA.cmake:42]: Done

**Review by PatKamin — APPROVED** [review]

**Comment by ldorau** [comment]
@bratpiorka @pbalcer re-review please

**Comment by lukaszstolarczuk** [comment]
random CI failure or we broke something in L0 provider...? 😉

**Comment by bratpiorka** [comment]
> random CI failure or we broke something in L0 provider...? 😉

only 2 tests fails:
 [  FAILED  ] urEnqueueEventsWaitMultiDeviceMTTest.EnqueueWaitSingleQueueMultiOps/MultiThread
 [  FAILED  ] urEnqueueKernelLaunchIncrementMultiDeviceMultiThreadTest.Success/NoUseEventsNoQueuePerThread

**Comment by ldorau** [comment]
@bratpiorka @lukaszstolarczuk All tests passed: `4 skipped and 73 successful checks`

**Comment by lukaszstolarczuk** [comment]
ref. https://github.com/intel/llvm/pull/16007
on intel/llvm testing two e2e tests failed:

https://github.com/intel/llvm/actions/runs/11721056696/job/32648261452?pr=16007#step:22:4415

**Comment by bratpiorka** [comment]
@ldorau please rebase

**Comment by ldorau** [comment]
> @ldorau please rebase

@bratpiorka Done

**Comment by lukaszstolarczuk** [comment]
> ref. [intel/llvm#16007](https://github.com/intel/llvm/pull/16007) intel/llvm testing

I used the new, rebased version and the testing passed on Friday - LGTM 👍

**Pull request merged**